airflow/example_dags/tutorial_etl_dag.py

```diff
@@ -0,0 +1,134 @@
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+### ETL DAG Tutorial Documentation
+This ETL DAG is compatible with Airflow 1.10.x (specifically tested with 1.10.12) and is referenced
+as part of the documentation that goes along with the Airflow Functional DAG tutorial located
+[here](https://airflow.apache.org/tutorial_functional_dag.html)
+"""
+# [START tutorial]
+# [START import_module]
+from datetime import timedelta
+import json
+
+# The DAG object; we'll need this to instantiate a DAG
+from airflow import DAG
+# Operators; we need this to operate!
+from airflow.operators.python_operator import PythonOperator
+from airflow.utils.dates import days_ago
+
+# [END import_module]
+
+# [START default_args]
+# These args will get passed on to each operator
+# You can override them on a per-task basis during operator initialization
+default_args = {
+    'owner': 'airflow',
+    'depends_on_past': False,
+    'email': ['airflow@example.com'],
+    'email_on_failure': False,
+    'email_on_retry': False,
+    'retries': 1,
+    'retry_delay': timedelta(minutes=5),
+}
+# [END default_args]
+
+# [START instantiate_dag]
+with DAG(
+    'tutorial_etl_dag',
+    default_args=default_args,
+    description='ETL DAG tutorial',
+    schedule_interval=timedelta(days=1),
+    start_date=days_ago(2),
+    tags=['example'],
+) as dag:
+# [END instantiate_dag]
+# [START documentation]
+    dag.doc_md = __doc__
+# [END documentation]
+
+# [START extract_function]
+    def extract(**kwargs):
+        ti = kwargs['ti']
+        data_string = u'{"1001": 301.27, "1002": 433.21, "1003": 502.22}'
+        ti.xcom_push('order_data', data_string)
+# [END extract_function]
+
+# [START transform_function]
+    def transform(**kwargs):
+        ti = kwargs['ti']
+        extract_data_string = ti.xcom_pull(task_ids='extract', key='order_data')
+        order_data = json.loads(extract_data_string)
+
+        total_order_value = 0
+        for value in order_data.values():
+            total_order_value += value
+
+        total_value = {"total_order_value": total_order_value}
+        total_value_json_string = json.dumps(total_value)
+        ti.xcom_push('total_order_value', total_value_json_string)
+# [END transform_function]
+
+# [START load_function]
+    def load(**kwargs):
+        ti = kwargs['ti']
+        total_value_string = ti.xcom_pull(task_ids='transform', key='total_order_value')
+        total_order_value = json.loads(total_value_string)
+
+        print(total_order_value)
+# [END load_function]
+
+# [START extract_task]
+    extract_task = PythonOperator(
+        task_id='extract',
+        python_callable=extract,
+    )
```

Comment:
We should use `@task()` decorator instead of `PythonOperator`, that was the main point of functional DAGs